src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -136,6 +136,18 @@ public bool HasValue
             }
         }
 
+        /// <summary>
+        /// Gets a <see cref="ReadOnlySpan{T}"/> from the current <see cref="StringSegment"/>.
+        /// </summary>
+        /// <returns>The <see cref="ReadOnlySpan{T}"/> from this <see cref="StringSegment"/>.</returns>
+        public ReadOnlySpan<char> AsSpan() => Buffer.AsReadOnlySpan(Offset, Length);
```

Comment:
Should this be AsReadOnlySpan?

Reply:
Remember, we are renaming these APIs on String to AsSpan and AsMemory, i.e. read-only being implied.

Reply:
Ok so this PR is good to go then.

Reply:
I think https://github.com/dotnet/corefx/pull/27328 is the mentioned rename. Version `4.5.0-preview2-26227-01` of `System.Memory` already has these renamed methods. Shall I submit an updated PR to be up to date with this -- or create an issue to track the renaming here?